### DISCUSSION TITLE
CAGR is not accurate - need to update to take the

### DISCUSSION THREAD

**Discussion by wpearce1** [ranaroussi/quantstats#311] [Ideas]
The current way to calculate CAGR is incorrect. The years component should be:

total_days = len(df.index)

years = total_days / 252. 

it looks like the way it is currently configured it counting weekends in the numerator but not the denominator.  this understates CAGR.

thanks

def cagr(returns, rf=0.0, compounded=True, periods=252):
    """
    Calculates the communicative annualized growth return
    (CAGR%) of access returns

    If rf is non-zero, you must specify periods.
    In this case, rf is assumed to be expressed in yearly (annualized) terms
    """
    total = _utils._prepare_returns(returns, rf)
    if compounded:
        total = comp(total)
    else:
        total = _np.sum(total)

    **years = (returns.index[-1] - returns.index[0]).days / periods**

    res = abs(total + 1.0) ** (1.0 / years) - 1

    if isinstance(returns, _pd.DataFrame):
        res = _pd.Series(res)
        res.index = returns.columns

    return res

**Comment by SugoiKitsune** [comment]
There is another issue when trrying to analyze assets/portfolios which don't follow standard US trading days convention. If you have more portfolio values within one year actual dates, quantstats would infer period length based on 252. Hence actual 300 days of portfolio returns within one year would be assumed as 1.2 years which eventually lead to incorrect CAGR. This is based on v0.77
